---
title: C6310 | Dokumentace Microsoftu
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C6310
helpviewer_keywords:
- C6310
ms.assetid: e759eb63-883f-4c3e-bf2f-b924ff511405
caps.latest.revision: 15
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 8bbe62582490f45c5bc1a5a7eb6a2c59820a4074
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: cs-CZ
ms.lasthandoff: 11/16/2018
ms.locfileid: "51800741"
---
# <a name="c6310"></a>C6310
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

upozornění C6310: Neplatná konstanta ve filtru výjimky může způsobit neočekávané chování  
  
 Tato zpráva znamená, že byla zjištěna neplatná konstanta ve výrazu filtru strukturovanou obslužnou rutinou. Jsou konstanty definované pro použití ve výrazu filtru strukturovanou obslužnou rutinou:  
  
- `EXCEPTION_CONTINUE_EXECUTION`  
  
- `EXCEPTION_CONTINUE_SEARCH`  
  
- `EXCEPTION_EXECUTE_HANDLER`  
  
  Tyto hodnoty jsou definovány v modulu runtime záhlaví souboru excpt.h.  
  
  Použití konstantu, která není v předchozím seznamu může způsobit neočekávané chování.  
  
## <a name="example"></a>Příklad  
 Následující kód vygeneruje toto upozornění:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (EXCEPTION_ACCESS_VIOLATION)  
  {  
    puts("Exception Occurred");  
  }  
}  
```  
  
 Chcete-li opravit toto upozornění, použijte následující kód:  
  
```  
#include <excpt.h>  
#include <stdio.h>  
#include <windows.h>  
  
BOOL LimitExceeded();  
  
void fd( )  
{  
  __try   
  {  
    if (LimitExceeded())   
    {  
      RaiseException(EXCEPTION_ACCESS_VIOLATION,0,0,0);  
    }  
    else  
    {  
      // code   
    }  
  }   
  __except (GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
  {  
    puts("Exception Occurred");  
  }  
}  
```



